tests/conftest.py

```diff
@@ -124,5 +128,21 @@ async def inner_coroutine():
 
 
 @pytest.fixture
-def static_directory_provider(tmp_path: Path):
-    return StaticDirectoryProvider(directory_path=tmp_path)
+def static_filename_provider():
+    return StaticFilenameProvider("ophyd_async_tests")
+
+
+@pytest.fixture
+def static_path_provider_factory(tmp_path: Path):
+    def create_static_dir_provider_given_fp(fp: FilenameProvider):
+        return StaticPathProvider(fp, tmp_path)
+
+    return create_static_dir_provider_given_fp
+
+
+@pytest.fixture
+def static_path_provider(
+    static_path_provider_factory: callable,
+    static_filename_provider: FilenameProvider,
+):
+    return static_path_provider_factory(static_filename_provider)
```

Comment:
```suggestion
@pytest.fixture
def static_path_provider(tmp_path: Path, static_filename_provider: StaticFilenameProvider):
        return StaticPathProvider(static_filename_provider, tmp_path)
```

Do you need this factory function elsewhere? It doesn't seem to be required.